bcipy/acquisition/tests/test_client.py

```diff
@@ -54,6 +54,44 @@ def __init__(self, *args, **kwargs):
         self.mock_channels = ['ch' + str(i) for i in range(num_channels)]
         self.mock_data = list(mock_data(num_records, num_channels))
 
+    def test_acquisition_exceptions(self):
+        """Exception should be thrown if unable to connect to device or message not understood """
+        daq = DataAcquisitionClient(device=None,
+                                    processor=NullProcessor())
+        daq._is_streaming = False
+        with self.assertRaises(Exception):
+            daq.start_acquisition()
+
+        daq.cleanup()
+
+    def test_daq_with_no_buffer(self):
+        """get_data should return an empty list if daq._buf is None
+        data length should return 0
+         """
+        
+        device = _MockDevice(data=self.mock_data, channels=self.mock_channels)
+        daq = DataAcquisitionClient(device=device,
+                                    processor=NullProcessor())
+        daq.start_acquisition()
+        time.sleep(0.1)
+        daq.stop_acquisition()
+
+        daq._buf = None
+
+        #test get_data
+        data = daq.get_data()
+        self.assertEqual(data,[])
```

Comment:
linting needed. not sure what IDE you're using now but try to find something that helps with that. You can run flake8 in the terminal if you'd prefer: http://flake8.pycqa.org/en/latest/

Reply:
Installed and got through as many suggestions as I could. It's a great tool! I've made the changes and will request Dani's review.